File: Desktop/DSA/T94/binarysearch.py


#binary search

def binary_search(array,target):

    n=len(array)
    left=0
    right=n-1

    while left < right:

        mid=(left+right)//2

        if array[mid] == target:

            return  mid

        elif array[mid] < target:

            left +=1

        else:

            right -=1

    return  -1

arr = [37, 23, 0, 17, 12, 72, 31, 46, 100, 88, 54]
print(binary_search(arr,100))

File: Documents/ANUDIP JAVA/binarysearch.java

import  java.util.Arrays;

public class binarysearch {

    public  static  void  main(String[] args){

        int[] arr = {2, 3, 4, 10, 40, 50, 60};
        int target = 10;

        // Perform Binary Search
        int result = binarySearch(arr, target);

        // Print the result
        if (result == -1) {
            System.out.println("Element not found in the array.");
        } else {
            System.out.println("Element found at index: " + result);
        }
    }

    public  static int binarySearch(int[] arr, int target){

        int left = 0;
        int right = arr.length - 1;

        while ( left <= right){

            int mid = (left+right) / 2;

            if  ( arr[mid] == target){

                return  mid;
            }

            if (arr[mid] > target){

                right = mid -1;
            }
            else {

                left =mid +1;
                }
        }

        return -1;
    }
}
